gumbo_client/client_v2.py

```diff
@@ -0,0 +1,46 @@
+import os
+from google.oauth2 import service_account
+from google.auth.transport.requests import AuthorizedSession
+
+import pandas as pd
+
+base_url = 'https://client-api-dot-depmap-gumbo.uc.r.appspot.com'
+gumbo_configs_dir = os.path.expanduser('~') + "/.config/gumbo/"
+client_id_filename = 'iap_client_id.txt'
+credentials_filename = 'client-iap-auth-sa.json'
+
+
+class Client:
+    def __init__(self):
+        # Read secrets from file 
+        with open(gumbo_configs_dir + client_id_filename, 'r') as file:
+            client_id = file.read().rstrip()
+        creds = service_account.IDTokenCredentials.from_service_account_file(
+            gumbo_configs_dir + credentials_filename,
+            target_audience=client_id)
+        # Get an authed session token 
+        self.authed_session = AuthorizedSession(creds)
+
+    def get(self, table_name: str) -> pd.DataFrame:
+        url = f'{base_url}/table/{table_name}'
+        response = self.authed_session.request("GET", url)
+        response.raise_for_status()
+        return pd.read_json(response.json())
+
+    def update(self, table_name, new_df, delete_missing_rows=False, reason=None):
+        pass
```

Comment:
When making stubs, I like to define them as
```
raise NotImplemented()
```
so that if I ever forget that I never defined the stub, its explictly communicates that when I try calling the function.